packages/wrangler/src/inspect.ts

```diff
@@ -0,0 +1,526 @@
+  }, [localWebSocket, remoteWebSocket]);
+}
+
+// Credit: https://stackoverflow.com/a/2117523
+function randomId(): string {
+  return "xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx".replace(/[xy]/g, function (c) {
+    const r = (Math.random() * 16) | 0,
+      v = c == "x" ? r : (r & 0x3) | 0x8;
+    return v.toString(16);
+  });
+}
+
+/**
+ * This function converts a message serialised as a devtools event
+ * into aarguments uitable to be called by a console method, and
+ * then actually calls the method with those arguments. Effectively,
+ * we're just doing a little bit of the work of the devtools console,
+ * directly in the terminal.
+ */
+function logConsoleMessage(evt: Protocol.Runtime.ConsoleAPICalledEvent): void {
+  const args = [];
+  for (const ro of evt.args) {
+    switch (ro.type) {
+      case "string":
+      case "number":
+      case "boolean":
+      case "undefined":
+      case "symbol":
+      case "bigint":
+        args.push(ro.value);
+        break;
+      case "function":
+        args.push(`[Function: ${ro.description}]`);
+        break;
+      case "object":
+        if (!ro.preview) {
+          args.push(ro.description);
+        } else {
+          args.push(ro.preview.description);
+
+          switch (ro.preview.subtype) {
+            case "array":
+              args.push(
+                "[ " +
+                  ro.preview.properties
+                    .map(({ value }) => {
+                      return value;
+                    })
+                    .join(", ") +
+                  (ro.preview.overflow ? "..." : "") +
+                  " ]"
+              );
+
+              break;
+            case "map":
+              args.push(
+                "{\n" +
+                  ro.preview.entries
+                    .map(({ key, value }) => {
+                      return `  ${key.description} => ${value.description}`;
+                    })
+                    .join(",\n") +
+                  (ro.preview.overflow ? "\n  ..." : "") +
+                  "\n}"
+              );
+
+              break;
+            case "set":
+              args.push(
+                "{ " +
+                  ro.preview.entries
+                    .map(({ value }) => {
+                      return `${value.description}`;
+                    })
+                    .join(", ") +
+                  (ro.preview.overflow ? ", ..." : "") +
+                  " }"
+              );
+
+              break;
+            case "null":
+              args.push("null");
+              break;
+            case "node":
+            case "regexp":
+            case "date":
+            case "weakmap":
+            case "weakset":
+            case "iterator":
+            case "generator":
+            case "error":
+            case "proxy":
+            case "promise":
+            case "typedarray":
+            case "arraybuffer":
+            case "dataview":
+            case "webassemblymemory":
+            case "wasmvalue":
```

Comment:
I still need to add logic for printing out these types, but I opened up the PR early anyway.

Reply:
Is there a library we can use?

Reply:
couldn't find any